pkg/scheduler/framework/runtime/framework.go

```diff
@@ -429,6 +429,7 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 	for _, pl := range f.preFilterPlugins {
 		status = f.runPreFilterPlugin(ctx, pl, state, pod)
 		if !status.IsSuccess() {
+			status.FailBy(pl.Name())
```

Comment:
for the sake of completion, should we do that for other extension points, especially PostFilter and Premit?

Reply:
For PostFilter, probably no. As it's more reasonable to return the "original" failure (in PreFilter or Filter) and rely upon that to decide moving a Pod or not.

For Permit, I think Permit more fits the coscheduling-like case that may depend on events  of particular custom resource to make it re-schedulable. And right now we're not open on that yet. So my idea was to be conservative until we open the registration for CRD.
However, I'm open to discussion. It seems enabling it is also harmless. It's just we should not wrap the permitStatus as plain error ([1], [2]) - otherwise the failedPlugin info gets lost. WDYT?

[1]: https://github.com/kubernetes/kubernetes/blob/7315c1f6ddf37e41ae7826a042d057d5f594c10c/pkg/scheduler/scheduler.go#L535
[2]: https://github.com/kubernetes/kubernetes/blob/7315c1f6ddf37e41ae7826a042d057d5f594c10c/pkg/scheduler/scheduler.go#L561

Reply:
I guess my suggestion is to set the plugin name wherever deterministic, whether or not it will be used downstream is a different discussion. In RunPermitPlugins for example, we return when the first plugin fails, so might as well set the plugin name. For RunPostFilterPlugins this is already done in Merge.

Reply:
@ahg-g I added failedPlugin to Permit(). Please take a look.

Given the PermitPlugin may return in the following paths:

1. immediately returned by PermitPlugin
1. timed out by framework
1. rejected by (another) PermitPlugin in subsequent scheduling cycles

we may have to change the interface `Reject(msg)` to `Reject(plugin, msg)` (or probably `Reject(msg, plugin ...string)`, but doesn't make a big difference IMO)

Reply:
thanks.